ocaml/testsuite/tests/module-type-underscore/typing.ml

```diff
@@ -0,0 +1,91 @@
+(* TEST
+ expect;
+*)
+
+module M : sig
+  module type S = sig
+    type t
+    val foo : t -> t
+    val bar : t list -> t
+  end
+end = struct
+  module type S = _
+end ;;
+[%%expect {|
+Uncaught exception: Failure("underscore not implemented")
+
+|}]
+
+module M : sig
+  module type S = _
```

Comment:
Should this trigger the error "underscore not allowed in signature"?

Reply:
If this type error is hard to catch currently (without proper typing), that means the type error is imprecise, then maybe we don't want this type error at all (i.e., just raise whenever spot an underscore).